c11threads.h

```diff
@@ -23,6 +23,10 @@ Main project site: https://github.com/jtsiomb/c11threads
 
 #define ONCE_FLAG_INIT	PTHREAD_ONCE_INIT
 
+#ifdef __APPLE__
+#define PTHREAD_MUTEX_TIMED_NP PTHREAD_MUTEX_NORMAL
+#endif
```

Comment:
Instead of using `#ifdef __APPLE__` everywhere, I suggest we do it once in the beginning where we also define some kind of `_C11THREADS_NO_TIMED_MUTEX` macro, and then test for that one later on where appropriate. This will make it explicit that it's not just some generic macosx compatibility conditional code block, but specifically that we're working around the fact that we don't have timed mutexes. If later on we find out that another platform doesn't support timed mutexes, we could just define the same macro again at the beginning.